### ISSUE TITLE
Accessing url is slow 

### ISSUE THREAD

**Issue by plribeiro3000** [fog/fog-google#43]
So I'm having a problem where If I try to access say `image.url` it takes a little too long. For example calling `url` on 100 images takes around 28 seconds when using Google Storage but using S3 with the same function call takes around half a second.

This might be related to this [CarrierWave issue](https://github.com/carrierwaveuploader/carrierwave/issues/261). However, the problem they were having was for both Google Storage and S3. @geemus mentioned in that issue to use `public_url` instead but this is still slow compared to using **fog** + S3. 

This might not be a **fog** issue but a Google Storage issue. 

As a workaround I'm currently just hardcoding the url to my Google Storage API and appending `image.path` to that - much faster.

**Comment by plribeiro3000** [comment]
The original issue was reported by @0x414f at fog/fog#3339.

cc/ @0x414f @geemus

**Comment by ihmccreery** [comment]
From @geemus:

> Both S3 and Google are setup to be slightly conservative here (in that they will try to ensure the file is publicly accessible with an API call, rather than returning a possibly erroneous file). It may just be that the calls that S3 does for this vs the Google ones return faster. See: https://github.com/fog/fog/blob/master/lib/fog/google/models/storage/file.rb#L94
If you are comfortable without that check, you can/should cut to the chase and do something similar to what you are doing I suspect. I think in some cases that is best anyway, but in the general case, the more cautious version seemed reasonable (especially since it can be worked around pretty easily). What do you think? Thanks!

**Comment by geemus** [comment]
Yeah, I think you can just build the url. If I remember correctly, by default we try to ensure the file exists (which is part of the slow down).

**Comment by Temikus** [comment]
@plribeiro3000 When you have time, can you verify whether this is still relevant with  
JSON API?

**Comment by plribeiro3000** [comment]
@Temikus Dunno. I just ported the original issue from the main `fog` repo.

I remember there was an issue for this same bahaviour on aws storage a while ago. Not sure what happened.

Maybe @geemus remember of it?

**Comment by geemus** [comment]
It may boil down to whether it accesses stuff on the object or calls something that just generates the url. In the S3 case it accessed model stuff (until we explicitly changed it to not do so in this context).

**Comment by HarlemSquirrel** [comment]
Hey guys, I'm new to this gem and cloud storage in general. I'm working on adding file uploads to my rails app and noticed this slow down with Google. I was able to speed things up a bit with [rails low-level caching](http://guides.rubyonrails.org/caching_with_rails.html#low-level-caching). 
```ruby
def cached_file_url
  Rails.cache.fetch("#{cache_key}/file_url") { file_url }
end
```
